src/MessageQuery.php

```diff
@@ -294,6 +297,10 @@ protected function fetch(Collection $messages): array
         return $this->connection()->fetch($fetch, $uids->all())->mapWithKeys(function (UntaggedResponse $response) {
             $data = $response->tokenAt(3);
 
+            if (!$data instanceof ListData) {
```

Comment:
I'm not sure as of right now. How you approach Exceptions. This is more or less a third approach. For the same problem. -> method returns parent, but specific child is expected

Reply:
Yea I think this is a good approach -- best to throw an exception here like you've implemented 👍